src/cpp/common/tls_certificate_provider.cc

```diff
@@ -23,6 +23,8 @@
 #include <grpc/grpc_security.h>
 #include <grpcpp/security/tls_certificate_provider.h>
 
+#include "src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.h"
```

Comment:
I think here we'd prefer to alias the new APIs in `grpcpp/security/tls_certificate_provider.h` if possible, rather that bring in the `src/core` import. And I believe we want this API exposed in the`grpcpp` include header so it can be used.
See the CRL Provider APIs as an exampe - https://github.com/grpc/grpc/blob/a615c505d1adea224a4b303f92f48fb60311a6c3/include/grpcpp/security/tls_crl_provider.h#L29-L34

Reply:
This works in the CRL provider case because there is no C++ layer implementation - it's shared with the C-core implementation. In this case, there is already a `FileWatcherCertificateProvider` defined in `tls_crl_provider.{h,cc}`, so I'm not sure we can do this without converging the two implementations.

Reply:
I think you can add a similar looking `::Create` to the definition in `tls_certificate_provider` that ends up wrapping the C-Core implementation or however you want to do it?

Reply:
Discussed offline